substrate/primitives/consensus/beefy/src/lib.rs

```diff
@@ -71,18 +71,23 @@ pub type BeefySignatureHasher = sp_runtime::traits::Keccak256;
 /// A trait bound which lists all traits which are required to be implemented by
 /// a BEEFY AuthorityId type in order to be able to be used in BEEFY Keystore
 pub trait AuthorityIdBound:
-	Codec
-	+ Debug
-	+ Clone
-	+ AsRef<[u8]>
-	+ ByteArray
-	+ AppPublic
-	+ AppCrypto
-	+ RuntimeAppPublic
-	+ Display
-	+ BeefyAuthorityId<BeefySignatureHasher>
+	Ord + AppPublic + Display + BeefyAuthorityId<BeefySignatureHasher>
```

Comment:
Do we need this trait ? Could we reuse `BeefyAuthorityId` instead ?

Reply:
I really don't know how to implement the black magic you suggested to get rid of `RuntimeApp::Signature : Send + Sync` with one trait.  I think I just followed your/the github issue order. Otherwise I need to bound `<Self as RuntimeApp>::Signature: Self::BoundedSignature`  which at that point the magic might as well break. If you know how to do that with one trait, then I'm happy to get rid of the second trait.

Reply:
I implemented you exact suggestion here: https://github.com/paritytech/polkadot-sdk/pull/1816/files/f75e3a21ce9d85f889410886522f06eedd27dad3#r1531880388